python/featomic/tests/clebsch_gordan/cartesian_spherical.py

```diff
@@ -198,3 +200,256 @@ def test_cartesian_to_spherical_errors(cartesian):
             components=["xyz_1", "xyz_2", "xyz_3"],
             keep_l_in_keys=False,
         )
+
+
+def _l2_components_from_matrix(A):
+    """
+    *The* reference equations for projecting a cartesian rank 2 tensor to the
+    irreducible spherical component with lambda = 2.
+    """
+    A = A.reshape(3, 3)
+
+    l2_A = np.empty((5,))
+    l2_A[0] = (A[0, 1] + A[1, 0]) / 2.0
+    l2_A[1] = (A[1, 2] + A[2, 1]) / 2.0
+    l2_A[2] = (2.0 * A[2, 2] - A[0, 0] - A[1, 1]) / ((2.0) * np.sqrt(3.0))
+    l2_A[3] = (A[0, 2] + A[2, 0]) / 2.0
+    l2_A[4] = (A[0, 0] - A[1, 1]) / 2.0
+
+    return l2_A * np.sqrt(2)
+
+
+def _l3_components_from_matrix(A):
+    """
+    *The* reference equations for projecting a cartesian rank 3 tensor to the
+    irreducible spherical component with lambda = 3.
+    """
+    A = A.reshape(3, 3, 3)
+
+    l3_A = np.empty((7,))
+
+    # -3
+    l3_A[0] = (A[0, 1, 0] + A[1, 0, 0] + A[0, 0, 1] - A[1, 1, 1]) / 2.0
+
+    # -2
+    l3_A[1] = (
+        A[0, 1, 2] + A[1, 0, 2] + A[1, 2, 0] + A[2, 1, 0] + A[0, 2, 1] + A[2, 0, 1]
+    ) / np.sqrt(6)
+
+    # -1
+    l3_A[2] = (
+        4.0 * A[1, 2, 2]
+        + 4.0 * A[2, 1, 2]
+        + 4.0 * A[2, 2, 1]
+        - 3.0 * A[1, 1, 1]
+        - A[0, 0, 1]
+        - A[0, 1, 0]
+        - A[1, 0, 0]
+    ) / np.sqrt(60)
+
+    # 0
+    l3_A[3] = (
+        2.0 * A[2, 2, 2]
+        - A[0, 2, 0]
+        - A[2, 0, 0]
+        - A[0, 0, 2]
+        - A[1, 2, 1]
+        - A[2, 1, 1]
+        - A[1, 1, 2]
+    ) / np.sqrt(10)
+
+    # 1
+    l3_A[4] = (
+        4.0 * A[0, 2, 2]
+        + 4.0 * A[2, 0, 2]
+        + 4.0 * A[2, 2, 0]
+        - 3.0 * A[0, 0, 0]
+        - A[1, 1, 0]
+        - A[0, 1, 1]
+        - A[1, 0, 1]
+    ) / np.sqrt(60)
+
+    # 2
+    l3_A[5] = (
+        A[0, 0, 2] + A[0, 2, 0] + A[2, 0, 0] - A[1, 1, 2] - A[1, 2, 1] - A[2, 1, 1]
+    ) / np.sqrt(6)
+
+    # 3
+    l3_A[6] = (A[0, 0, 0] - A[1, 1, 0] - A[0, 1, 1] - A[1, 0, 1]) / 2.0
+
+    return l3_A
+
+
+@pytest.mark.parametrize("cg_backend", ["python-dense", "python-sparse"])
+def test_cartesian_to_spherical_rank_2_by_equation(cg_backend):
+    """
+    Tests cartesian_to_spherical for a random rank-2 tensor by comparing the result to
+    the result from *the* reference equation.
```

Comment:
```suggestion
def test_cartesian_to_spherical_rank_2(cg_backend):
    """
    Tests cartesian_to_spherical for a random rank-2 tensor by comparing the result to
    the result from the reference equation.
```

Reply:
There's already a unit test of that name!